File: src/main/java/thkoeln/dungeon/game/domain/GameRepository.java
package thkoeln.dungeon.game.domain;

import org.springframework.data.repository.CrudRepository;

import java.util.List;
import java.util.UUID;

public interface GameRepository extends CrudRepository<Game, UUID> {
    public List<Game> findByGameId( UUID gameId );
    public boolean existsByGameId( UUID gameId );
    public List<Game> findAllByGameStatusEquals( GameStatus gameStatus );
    public List<Game> findAllByGameStatusBetween( GameStatus gameStatus1, GameStatus gameStatus2 );
    public List<Game> findAll();
}

File: src/main/java/thkoeln/dungeon/game/domain/GameRepository.kt
package thkoeln.dungeon.game.domain

import org.springframework.data.repository.CrudRepository
import java.util.*

interface GameRepository : CrudRepository<Game?, UUID?> {
    fun findByGameId(gameId: UUID?): List<Game?>?
    fun existsByGameId(gameId: UUID?): Boolean
    fun findAllByGameStatusEquals(gameStatus: GameStatus?): List<Game?>?
    fun findAllByGameStatusBetween(gameStatus1: GameStatus?, gameStatus2: GameStatus?): List<Game?>?
    override fun findAll(): List<Game?>
    fun save(game: Game?)
}